### PR TITLE
[existing-filter] Filter already-open worktrees from Existing tab

### PR DESCRIPTION
**Pull request by mcaden**
The Existing tab in the Add Worktree dialog now filters out worktrees that are already loaded as tabs.

Changes:
- Added a useMemo that computes loaded worktree paths from the tab store and excludes them from the list
- Shows a distinct empty-state message (All worktrees are already open) when worktrees exist but are all loaded, vs No worktrees found when none exist on disk
- Two new tests cover both filtering and the all-open message scenarios

### CODE REVIEW

File: src/components/NewSessionDialog.tsx [hunk 1]
```diff
@@ -146,6 +147,12 @@ export function NewSessionDialog(): JSX.Element | null {
     return validateWorktreeName(trimmed);
   }, [newName]);
 
+  // Filter out worktrees that are already loaded as tabs.
+  const availableWorktrees = useMemo(() => {
+    const loadedPaths = new Set(openTabs.map((t) => t.path));
+    return worktrees.filter((w) => !loadedPaths.has(w.path));
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src/components/NewSessionDialog.tsx]: :robot: AI agent reply (acting for @mcaden):

Done in e8f3b4a3. Replaced exact string equality with `pathsEqual()` from `worktree-paths.ts` which normalizes separators to `/`, trims trailing slashes, and case-folds on Windows-like paths. Added unit tests for `pathsEqual` and an integration test exercising mixed-separator filtering.

File: src/lib/worktree-paths.ts [hunk 2]
```diff
@@ -16,13 +16,21 @@ function isWindowsLikePath(p: string): boolean {
...
   // backslashes in literal POSIX file names are rare, and the inputs to
   // this helper come from `git worktree list --porcelain` and our own
   // workspace-root config — neither produces such names in practice.
-  return p.replace(/\\/g, '/').replace(/\/+$/, '');
+  const slashed = p.replace(/\\/g, '/');
+  // Preserve root paths: `/`, `C:/`, `//server` — stripping their trailing
+  // slash would change semantics (empty string or `C:` = CWD on that drive).
+  if (slashed === '/') return '/';
+  if (/^[A-Za-z]:\/$/.test(slashed)) return slashed;
+  // Strip trailing slashes without regex quantifiers (avoids SonarCloud ReDoS false positive).
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src/lib/worktree-paths.ts]: :robot: AI agent reply (acting for @mcaden):

Done in 5866f0e1. `isInsideWorktreesDir()` now checks whether the normalized root already ends with `/` before inserting the separator, avoiding the double-slash. Also hardened `normalize()` to collapse all-slash inputs (`////` → `/`) and multi-slash drive roots (`C:////` → `C:/`). Added tests for these edge cases.

File: src/lib/worktree-paths.ts [hunk 3]
```diff
@@ -16,13 +16,21 @@ function isWindowsLikePath(p: string): boolean {
...
+  let end = slashed.length;
+  while (end > 0 && slashed[end - 1] === '/') end--;
+  return end === 0 ? slashed : slashed.slice(0, end);
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src/lib/worktree-paths.ts]: :robot: AI agent reply (acting for @mcaden):

Done in 5866f0e1. `normalize()` now treats any all-slash input as POSIX root (`/`) and broadens drive-root detection to `/^[A-Za-z]:\/+$/` so `C:////` normalizes to `C:/`. Added unit tests for both cases.

File: src/components/NewSessionDialog.tsx [hunk 4]
```diff
@@ -146,6 +206,11 @@ export function NewSessionDialog(): JSX.Element | null {
     return validateWorktreeName(trimmed);
   }, [newName]);
 
+  // Filter out worktrees that are already loaded as tabs.
+  const availableWorktrees = useMemo(() => {
+    return worktrees.filter((w) => !openTabs.some((t) => pathsEqual(t.path, w.path)));
+  }, [worktrees, openTabs]);
+
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src/components/NewSessionDialog.tsx:215]: :robot: AI agent reply (acting for @mcaden):

Done in 5866f0e1. Replaced `useWorktreeTabs()` with a conditional `useWorktreeTabStore(useShallow(...))` selector that returns `[]` when `isOpen` is false, so tab store mutations no longer trigger re-renders while the dialog is hidden.

File: src/lib/worktree-paths.ts [hunk 5]
```diff
@@ -16,13 +16,21 @@ function isWindowsLikePath(p: string): boolean {
...
-  return p.replace(/\\/g, '/').replace(/\/+$/, '');
+  const slashed = p.replace(/\\/g, '/');
+  // Drive root with any number of trailing slashes (e.g. `C:/`, `C:////`) → `X:/`.
+  if (/^[A-Za-z]:\/+$/.test(slashed)) return slashed[0] + ':/';
+  // Strip trailing slashes without regex quantifiers (avoids SonarCloud ReDoS false positive).
+  let end = slashed.length;
+  while (end > 0 && slashed[end - 1] === '/') end--;
+  // All slashes stripped → input was all slashes → POSIX root.
+  if (end === 0) return '/';
+  return slashed.slice(0, end);
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src/lib/worktree-paths.ts]: :robot: AI agent reply (acting for @mcaden):

Done in a5965663. Added an early return for empty strings so `normalize('')` returns `''` instead of falling through to the root-path case. Added test asserting `pathsEqual('', '/') === false`.

File: src/components/NewSessionDialog.tsx [hunk 6]
```diff
@@ -38,10 +39,71 @@ function deriveLabel(path: string): string {
...
+  }
+  return (
+    <ul className="themed-scrollbar mb-2 max-h-48 overflow-y-auto rounded border border-slate-200 dark:border-slate-700">
+      {availableWorktrees.map((w) => (
+        <li key={w.path}>
+          <button
+            type="button"
+            onClick={() => onSelect(w)}
+            className={`flex w-full items-center justify-between gap-2 px-3 py-2 text-left text-sm hover:bg-slate-100 dark:hover:bg-slate-700 ${
+              selectedPath === w.path ? 'bg-sky-100 dark:bg-sky-900' : ''
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src/components/NewSessionDialog.tsx]: :robot: AI agent reply (acting for @mcaden):

Done in a5965663. Replaced `selectedPath === w.path` with `pathsEqual(selectedPath, w.path)` (null-guarded) so selection highlighting is cross-platform consistent.

File: src/components/NewSessionDialog.tsx [hunk 7]
```diff
@@ -12,14 +12,15 @@
 // in tests covers `showModal`/`close`).
 
 import { useEffect, useMemo, useRef, useState } from 'react';
+import { useShallow } from 'zustand/shallow';
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src/components/NewSessionDialog.tsx]: :robot: AI agent reply (acting for @mcaden):

Done in a220ae56. Changed import to `zustand/react/shallow` to match the established pattern in config-store, session-store, sub-session-store, and WorktreePrepBanner.

File: src/components/NewSessionDialog.tsx [hunk 8]
```diff
@@ -146,6 +208,11 @@ export function NewSessionDialog(): JSX.Element | null {
     return validateWorktreeName(trimmed);
   }, [newName]);
 
+  // Filter out worktrees that are already loaded as tabs.
+  const availableWorktrees = useMemo(() => {
+    return worktrees.filter((w) => !openTabs.some((t) => pathsEqual(t.path, w.path)));
+  }, [worktrees, openTabs]);
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src/components/NewSessionDialog.tsx:214]: :robot: AI agent reply (acting for @mcaden):

Valid concern. However, this is a backend change to `worktrees_list` (making it return canonicalized paths) and is out of scope for this UI-filtering PR. The current approach already handles the practical cases (separator/casing/trailing-slash differences). Symlink canonicalization would need to be addressed holistically across all commands that return paths. Will file a follow-up issue.

[the remark this answers is not in the document]
Comment (mcaden) [reply src/components/NewSessionDialog.tsx:214]: :robot: AI agent reply (acting for @mcaden):

Done in fa1afc71. Backend `list_worktrees` now canonicalizes each path via `dunce::canonicalize` after parsing `git worktree list --porcelain` output, matching the canonicalization applied to tab paths by `validate_worktree`. Symlinked worktree paths will now compare correctly in the UI filter.
